server/statistics/hot_cache.go

```diff
@@ -123,40 +59,36 @@ func (w *HotCache) CheckReadPeerSync(peer *core.PeerInfo, region *core.RegionInf
 }
 
 // CheckWriteAsync puts the flowItem into queue, and check it asynchronously
-func (w *HotCache) CheckWriteAsync(item *FlowItem) {
-	w.writeFlowQueue <- item
+func (w *HotCache) CheckWriteAsync(task FlowItemTask) {
+	w.writeFlowQueue <- task
 }
 
 // CheckReadAsync puts the flowItem into queue, and check it asynchronously
-func (w *HotCache) CheckReadAsync(item *FlowItem) {
-	w.readFlowQueue <- item
+func (w *HotCache) CheckReadAsync(task FlowItemTask) {
+	w.readFlowQueue <- task
 }
 
 // Update updates the cache.
 func (w *HotCache) Update(item *HotPeerStat) {
 	switch item.Kind {
 	case WriteFlow:
-		w.writeFlow.Update(item)
+		update(item, w.writeFlow)
 	case ReadFlow:
-		w.readFlow.Update(item)
-	}
-
-	if item.IsNeedDelete() {
-		w.incMetrics("remove_item", item.StoreID, item.Kind)
-	} else if item.IsNew() {
-		w.incMetrics("add_item", item.StoreID, item.Kind)
-	} else {
-		w.incMetrics("update_item", item.StoreID, item.Kind)
+		update(item, w.readFlow)
 	}
 }
 
 // RegionStats returns hot items according to kind
 func (w *HotCache) RegionStats(kind FlowKind, minHotDegree int) map[uint64][]*HotPeerStat {
 	switch kind {
 	case WriteFlow:
-		return w.writeFlow.RegionStats(minHotDegree)
+		task := newCollectRegionStatsTask(minHotDegree)
+		w.writeFlowQueue <- task
+		return <-task.(*collectRegionStatsTask).ret
 	case ReadFlow:
-		return w.readFlow.RegionStats(minHotDegree)
+		task := newCollectRegionStatsTask(minHotDegree)
+		w.readFlowQueue <- task
+		return <-task.(*collectRegionStatsTask).ret
 	}
 	return nil
 }
```

Comment:
The other interface like `IsRegionHot` also need to handle.

Reply:
updated.